doc/documentation_workflow.rst

```diff
@@ -0,0 +1,152 @@
+Documentation workflow
+######################
+
+What you need to know
++++++++++++++++++++++
+
+We use **Sphinx** to generate documentation and **Read the Docs** to publish it.
+
+The documentation for NEST is contained in the ``docs`` directory within the `NEST source code repository <https://github.com/nest/nest-simulator>`_ on GitHub.
+
+Using **GitHub** allows the NEST simulator documentation to live alongside its code. Our developers benefit from smooth version control, with each release of NEST having its own documentation.
+
+.. image:: _static/img/documentation_workflow.png
+  :width: 300
+  :alt: Alternative text
+
+Changing the documentation
+++++++++++++++++++++++++++
+
+If you notice something out of place, you can submit an `issue <https://github.com/nest/nest-simulator/issues>`_ in our GitHub repository.
+
+You can also make changes directly to your forked copy of the `NEST source code repository <https://github.com/nest/nest-simulator>`_ and create a `pull request <https://github.com/nest/nest-simulator/pulls>`_. Just follow the workflow below!
+
+Setting up your environment
++++++++++++++++++++++++++++
+
+To keep things simple, we have created a conda environment for you. Installing it will enable you to smoothly generate documentation for NEST.
+
+If you are using Linux and want to install a full development environment:
+
+1. Install conda (we recommend `miniconda <https://docs.conda.io/en/latest/miniconda.html#>`_).
+
+2. Switch to the ``doc`` folder in the source directory:
+
+.. code-block:: bash
+
+    cd git/nest-simulator/doc
+
+3. Create and activate the environment:
+
+.. code-block:: bash
+
+   conda env create -f conda_environment.yml
+   conda update -n base -c defaults conda
+   conda activate doc
+
+4. If you want to deactivate or delete the build environment:
+
+.. code-block:: bash
+
+   conda deactivate
+   conda remove --name doc --all
+
+Generating with Sphinx
+++++++++++++++++++++++
+
+Now that you activated your environment, you can generate HTML files using `Sphinx <https://www.sphinx-doc.org/en/master/>`_.
+
+Sphinx uses reStructuredText. To learn more on how to use it, check out this `quick reference <https://docutils.sourceforge.io/docs/user/rst/quickref.html>`_.
+
+Editing and creating pages
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. You can edit and/or add ``.rst`` files in the ``doc`` directory using your editor of choice.
+
+2. If you create a new page, open ``index.rst`` in the ``doc`` directory and add the file name under ``.. toctree::``. This will ensure it appears on the NEST simulator documentation's table of contents.
+
+3. Save your changes.
+
+Rendering HTML
```

Comment:
Should we move the generating page up, before the modification info?

Reply:
I think the ``Generating documentation with Sphinx`` section fits well where it is. How about we move the sentence on Sphinx using reStructured text up? That could be added to ``What you need to know``.

Reply:
@sarakonradi OK.

Reply:
Done.

Reply:
The "Generating documentation with Sphinx" section is in a good place. Two thoughts on this though:

The sentence about reStructured text is fine up there, but I would repeat it with the the link also at the beginning of the "editing and creating pages" section. That's where you need the link and one may have skipped the colorful intro... or be too lazy to scroll all the way back up again.

I would also suggest to exchange the sections "Editing and creating pages" and "Rendering HTML". Reading a section on how to generate documentation I stumbled over "editing files" thinking "why do I need to edit files to build the documentation?. The section "Rendering HTML" is how you build the documentation yourself. Actually, going in and changing something is (read "should be") always the second step. Otherwise it is not possible to understand if a broken rendering is due to a problem introduced by the editing or due to other issues in the environment or a prior issues in the code base. When reading only the Rending HTML section, it sounds strange to have point 4. saying "add further changes", when points 1.-3. didn't do any.

My suggestion would be to exchange the sections, add a new point to editing and creating pages: "5. re-render the documentation as described above" and remove the point 4 from the rendering HTML section.

Reply:
You are right, I see your point. Done.